src/org/thoughtcrime/securesms/ShareActivity.java

```diff
@@ -138,6 +138,7 @@ private boolean needsFilePermission(List<Uri> uris) {
   private void requestPermissionForFiles(List<Uri> streamExtras) {
     Permissions.with(this)
             .request(Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.READ_EXTERNAL_STORAGE)
```

Comment:
`WRITE_EXTERNAL_STORAGE` is ".alwaysGrantOnSdk30()" in other places, does this mean that it will be requested for SDK 30 here? what are the implications?

Reply:
i assume, that should be `alwaysGrantOnSdk30()` as well :)

Reply:
hm, not sure. in contrast to the other places, this uses some `file:` protocol, that may indeed require these rights (similar to the camera-roll, that also requires READ_EXTERNAL_STORAGE between sdk30..sdk32 (implicitly requested by galleryPermissions()). let's leave this as it was before (so, only change this for sdk33 where the permissions are gone.

might be, that we would better use galleryPermissions() for sdk33 here - but i anyway assume that "files" are playing less and less roles on newer androids, so, probably everything's fine. if not - we can iterate, if things are wrong on testing